tools/autograd/gen_autograd.py

```diff
@@ -42,21 +42,28 @@
     'as_strided': 'self',
     'diagonal': 'self',
     'expand': 'self',
+    'expand_as': 'self',
     'narrow': 'self',
     'permute': 'self',
+    'reshape': 'self',
+    'reshape_as': 'self',
     'select': 'self',
     'slice': 'self',
     'squeeze': 'self',
     't': 'self',
     'transpose': 'self',
     'unfold': 'self',
     'unsqueeze': 'self',
+    'flatten': 'flatten',
```

Comment:
This is a good catch. Be sure to add the PR as BC-breaking and add a note in the PR description stating that `flatten` will now properly detect it's output as a view. So inplace ops on the result will now properly be tracked and the proper backward graph will be created.

Reply:
Done, thanks.